src/app/zap-templates/templates/app/helper.js

```diff
@@ -28,14 +28,33 @@ const dbEnum       = require(zapPath + '../src-shared/db-enum.js')
 
 const StringHelper    = require('../../common/StringHelper.js');
 const ChipTypesHelper = require('../../common/ChipTypesHelper.js');
+const { Deferred }    = require('../../common/Deferred.js');
 
 zclHelper['isEvent'] = function(db, event_name, packageId) {
-    return queryEvents
-      .selectAllEvents(db, packageId)
+    if (!this._cacheEvents) {
+        this._cacheEvents = new Deferred();
+        queryEvents.selectAllEvents(db, packageId).then(events => {
+          this._cacheEvents.resolve(events);
+        });
+    }
+    return this._cacheEvents
       .then(events => events.find(event => event.name == event_name))
       .then(events => events ? 'event' : dbEnum.zclType.unknown);
 }
 
+zclHelper['isCommand'] = async function(db, command_name, packageId) {
```

Comment:
Nothing prevents a command having the same name as a type does, right?  What guarantees we won't have name collisions?

It would be better to just have a `commandStruct` helper or something that does specifically commands.

Reply:
https://github.com/project-chip/connectedhomeip/blob/e71be1c36ca713b5d9f12008097ec9e0f44e560f/src/app/zap-templates/templates/app/helper.js#L500 will throw if there is a collision in the names in the future.

If you really want a dedicated helper I can do it though but it was looking nice to have the same helper for all encodable/ decodable types !

Reply:
That would just make codegen fail for someone who would then have to choose between not using the spec names and trying to fix the codegen.

I would really prefer a separate helper for this.  I think it makes a lot more sense.